### ISSUE TITLE
[mqtt] Homie things with a large number of properties may fail with OFFLINE - COMMUNICATION_ERROR Did not receive all required topics

### ISSUE THREAD

**Issue by leifclaesson** [openhab/openhab-addons#5787]
I discovered this issue when troubleshooting [issue 5786](https://github.com/openhab/openhab2-addons/issues/5786)

## Current Behavior
With trace logging enabled on *org.eclipse.smarthome.io.transport.mqtt*, when trying to add a Homie device with many properties (10 or so, or more) then the added Homie device thing will show COMMUNICATION_ERROR Did not receive all required topics, even though all topics were present as retained MQTT messages.

Same [forum thread](https://community.openhab.org/t/mqtt-items-unreliable/77232/13) as issue #5786 but it is a separate bug.

## Expected Behavior
The topics were all there so I was expecting the Homie device thing to be ONLINE.

## Possible Solution
It could be a race condition in the binding. Trace logging slows the binding down due to a large number of regular expressions being matched. Perhaps the retained messages did not come through before a fixed timeout value.
It would be better if the state was refreshed as the new topics come in.

## Steps to Reproduce (for Bugs)
My own *homielib* which I just released on github may be helpful in reproducing the problem.
https://github.com/liket/homielib
It's a Homie v3.0.1 library for ESP8266.

Please note that after I knew how to reproduce bug #5786 I also disabled trace logging, and then this problem went away. However if there is a race condition it *can* still happen under other circumstances so it would be better to actualy fix the bug, also so that enabling logging don't make things behave differently because that can make troubleshooting nearly impossible.

Also, after this problem went away, a new issue occured: The following error messages appear in the log every time my Homie Device properties were updated by the ESP8266 running the library.

```
2019-07-01 18:47:07.287 [WARN ] [.core.thing.binding.BaseThingHandler] - Handler HomieThingHandler of thing mqtt:homie300:506c59a4 tried updating channel nodeid1#hum although the handler was already disposed.

2019-07-01 18:47:09.289 [WARN ] [.core.thing.binding.BaseThingHandler] - Handler HomieThingHandler of thing mqtt:homie300:506c59a4 tried updating channel nodeid1#precip although the handler was already disposed.

2019-07-01 18:47:11.291 [WARN ] [.core.thing.binding.BaseThingHandler] - Handler HomieThingHandler of thing mqtt:homie300:506c59a4 tried updating channel nodeid1#wind although the handler was already disposed.

2019-07-01 18:47:15.292 [WARN ] [.core.thing.binding.BaseThingHandler] - Handler HomieThingHandler of thing mqtt:homie300:506c59a4 tried updating channel nodeid1#hum although the handler was already disposed.

2019-07-01 18:47:17.290 [WARN ] [.core.thing.binding.BaseThingHandler] - Handler HomieThingHandler of thing mqtt:homie300:506c59a4 tried updating channel nodeid1#precip although the handler was already disposed.
```

It's probably not related but I'm writing it here instead of creating a third bug report in one night.

## Your Environment
openHAB 2.5.0M1 on RPi 3B+ with SSD
Mosquitto on the same machine

**Comment by J-N-K** [comment]
Is this still the case with latest snapshot?

**Comment by RayBa82** [comment]
I have the same issue with 3.0.0 snapshot. (Yes 3.0.0 is correct)
If i stop the external broker and restart the broker its the same issue, the devices are not coming online again. Restarting openhab fixes it.

In my setup everything runs on the same machine - OpenHAB, mosquitto and some spring boot microservices which are the homie devices.

Worst case is a power loss, the homie devices dont work after a reboot of the machine. They all show communication errors

**Comment by RayBa82** [comment]
In some rare cases the devices are online after reboot.
Sometimes a restart of openhab does not even help.

For me the homie devices is broken again. After a complete broken implementation in openhab 2.4, the snapshots did work stable for a long time, now in 2.5 it is broken again.....

**Comment by mjcumming** [comment]
@leifclaesson and @RayBa82, I think you are seeing the issue as described in this [thread](https://github.com/openhab/openhab-addons/issues/7252)

**Comment by mjcumming** [comment]
For me, this is fixed in 2.5.6~S131
